### PR TITLE
feat: --browsers option for default browsers

### PR DESCRIPTION
**Pull request by karlhorky**
Prerequisite (stacked on): #118

Closes microsoft/playwright#29893

### CODE REVIEW

File: src/generator.ts [hunk 1]
```diff
@@ -132,7 +147,8 @@ export class Generator {
         type: 'confirm',
         name: 'installPlaywrightBrowsers',
         message: `Install Playwright browsers (can be done manually via '${this.packageManager.npx('playwright', 'install')}')?`,
-        initial: true,
+        initial: true,    
+        skip: () => !!this.options.browsers,
```
Comment (karlhorky) [inline src/generator.ts:151]: skip the prompt question, defaulting to `true`, by using the `enquirer` prompt option `skip()`:

- https://github.com/enquirer/enquirer/issues/364#issuecomment-1991338065

File: src/generator.ts [hunk 2]
```diff
@@ -33,21 +33,36 @@ export type PromptOptions = {
...
+      : BooleanOptionName
+    : never,
+  'help'
+>;
 
 export class Generator {
   private packageManager: PackageManager;
 
-  constructor(private readonly rootDir: string, private readonly options: Partial<Record<CliArgumentKey, string[]>>) {
+  constructor(private readonly rootDir: string, private readonly options: Partial<Record<OptionName, string[]>>) {
```
Comment (karlhorky) [inline src/generator.ts:65]: rename variable for consistency

File: src/generator.ts [hunk 3]
```diff
@@ -33,21 +33,36 @@ export type PromptOptions = {
...
+export type Option = (typeof allOptions)[number][0];
+
+export type OptionName = Exclude<
+  Option extends `--${infer BooleanOptionName}`
+    ? BooleanOptionName extends `${infer NonBooleanOptionName}=<${string}>`
+      ? NonBooleanOptionName
+      : BooleanOptionName
+    : never,
+  'help'
+>;
```
Comment (karlhorky) [inline src/generator.ts:60]: use conditional types and `infer` types to avoid duplicating options in 2 places

File: src/cli.ts [hunk 4]
```diff
@@ -34,11 +48,25 @@ import { Generator } from './generator';
...
+  );
+  if (bothBooleanOptionsSpecified.length > 0) {
+    console.error(
+      `Cannot specify both of:
+  ${bothBooleanOptionsSpecified
+    .map(([positive, negative]) => `- ${positive} and ${negative}`)
+    .join('\n')}`,
+    );
     process.exit(1);
   }
```
Comment (karlhorky) [inline src/cli.ts:67]: prevent both boolean options from being passed (eg. both `--no-browsers` and `--browsers`) - will work for all future boolean options

File: src/generator.ts [hunk 5]
```diff
@@ -132,7 +147,8 @@ export class Generator {
         type: 'confirm',
         name: 'installPlaywrightBrowsers',
         message: `Install Playwright browsers (can be done manually via '${this.packageManager.npx('playwright', 'install')}')?`,
-        initial: true,
+        initial: true,    
+        skip: !!this.options.browsers,
```
Comment (karlhorky) [inline src/generator.ts:151]: skip the prompt question, defaulting to `true`, by using the `enquirer` prompt option `skip`:

- https://github.com/enquirer/enquirer#prompt-options

**Comment by karlhorky** [comment]
@mxschmitt thanks for the merge of #118 🙌 

This PR was stacked on #118, and since it's been merged, I've marked this PR as ready for review as well.

**Comment by karlhorky** [comment]
Merged from `main` and fixed conflicts again.

**Comment by karlhorky** [comment]
Since this is a pretty big PR, and contains other improvements such as [use conditional types and `infer` types to avoid duplicating options in 2 places](https://github.com/microsoft/create-playwright/pull/119#discussion_r1533067875), I think I'll split this out into multiple smaller PRs...

**Comment by mxschmitt** [comment]
Closing as per https://github.com/microsoft/playwright/issues/29893 since we didn't see enough activity on this feature.

**Comment by karlhorky** [comment]
@mxschmitt too bad, it's disappointing that small features are not considered in different ways. It's discouraging to my motivation, as a community member who wants to contribute.

**Pull request closed without merging**